### ISSUE TITLE
[v2] Android - setRoot doesn't update modals

### ISSUE THREAD

**Issue by jordanfloyd** [wix/react-native-navigation#4474]
### Issue Description

When calling Navigation.setRoot on Android, modals aren't dismissed.

### Steps to Reproduce / Code Snippets / Screenshots

```
Navigation.showModal({
  stack: {
    children: [{
      component: {
        name: 'ScreenName'
      }
    }]
  }
})

Navigation.setRoot({
    modals: [], //  with or without this, doesn't make a difference
    root: {
      stack: {
        children: [
          {
            component: {
              name: 'ScreenName',
              passProps: {
                ...props
              }
            }
          }
        ]
      }
    }
})
```
iOS dismisses any active modals, Android does not.

Also tangentially related, calling dismissAllModals on Android when there are none throws an error instead of being a no-op, which crashes the app.

---
### Environment
* React Native Navigation version: 2.2.1
* React Native version: 0.55.0
* Platform(s) (iOS, Android, or both?): Android
* Device info (Simulator/Device? OS version? Debug/Release?): Android Pixel 2 Android 8.1

**Comment by jordanfloyd** [comment]
This may have been an issue on my end. Closing for now until I can consistently reproduce.

**Comment by Collin3** [comment]
Is there a reason why this is closed? It still appears to be an issue

**Comment by eliaslecomte** [comment]
I agree this should be reopened!

**Comment by bitcrumb** [comment]
+1

We're seeing this issue as well. This should be reopened since behavior on Android is now different from iOS.

**Comment by guyca** [comment]
Hey guys, this is fixed in 2.24.0 👍
